Sources/BetterCodable/DefaultCodable.swift

```diff
@@ -46,4 +50,27 @@ public extension KeyedDecodingContainer {
             return DefaultCodable(wrappedValue: P.defaultValue)
         }
     }
+
+    func decode<P: BoolCodableStrategy>(_: DefaultCodable<P>.Type, forKey key: Key) throws -> DefaultCodable<P> {
+        do {
+            let value = try decode(Bool.self, forKey: key)
+            return DefaultCodable(wrappedValue: value)
+        } catch let error {
+            guard let decodingError = error as? DecodingError,
+                case .typeMismatch = decodingError else {
+                    return DefaultCodable(wrappedValue: P.defaultValue)
+            }
+            // Not sure if it is worth adding the Double, Float, and other cases.
+            // In reality backend should never return double, or floating point values for bools but hey we are here for a reason 🤷🏻‍♂️
```

Comment:
I think not in this case, since LosslessValue should now handle those conditions. We can remove this comment so future self is not confused 😄